Added muenv script for Python env activatation/deactivation

sam-grant opened the pull request:
Added script `muenv.sh` to easily activate and deactivate the Mu2e Python environment, followings Ray's recommendation. 

Demo:

```
[sgrant@mu2egpvm06 archive]$ source setupmu2e-art.sh 
[sgrant@mu2egpvm06 archive]$ muenv

    Usage: muse activate ENV_NAME [VERSION]

    Activates a Mu2e Python environment.
    
    Available environments:
      • ana     - Standard analysis environment
      • rootana - Analysis environment with pyROOT support
    
    Parameters:
      ENV_NAME - Required: Environment name ('ana' or 'rootana')
      VERSION  - Optional: Environment version (default: 'current')
    
    Examples:
      muenv ana
      muenv rootana 1.2.0
      
    The first example runs:
      source /cvmfs/mu2e.opensciencegrid.org/env/ana/current/bin/activate

    See https://mu2ewiki.fnal.gov/wiki/Elastic_Analysis_Facility_(EAF)#Change_log for version info 

    <command options>
        -h, --help  : print usage

[sgrant@mu2egpvm06 archive]$ muenv ana
Activating Mu2e Python environment: ana current
Run 'deactivate' to exit the environment
(current) [sgrant@mu2egpvm06 archive]$ deactivate
[sgrant@mu2egpvm06 archive]$ muenv rootana 1.2.0
Activating Mu2e Python environment: rootana 1.2.0
Run 'deactivate' to exit the environment
(1.2.0) [sgrant@mu2egpvm06 archive]$ deactivate
[sgrant@mu2egpvm06 archive]$ 
```

Note: I included the child `deactivate` function, since the in-built `deactivate` returns `Permission denied`. 

Please let me know what you think.

rlcee commented:
Very nice!
Do you understand the reason the builtin deactivate gives permission denied?  Does it try to write to the env dir?
When the script is sourced, it will leave the help function in the user environment, which I think is little better to leave uncluttered.  So move the help function to no function? Or add "unset -f  mu2envUsage"?
In the function in setupmu2e-art
source mu2env.sh 
will only work by searching PATH if a certain bash opt is set.  Please add a path like
/cvmfs/mu2e.opensciencegrid.org/bin/muenv.sh

sam-grant commented:
> Very nice! Do you understand the reason the builtin deactivate gives permission denied? Does it try to write to the env dir? When the script is sourced, it will leave the help function in the user environment, which I think is little better to leave uncluttered. So move the help function to no function? Or add "unset -f mu2envUsage"? In the function in setupmu2e-art source mu2env.sh will only work by searching PATH if a certain bash opt is set. Please add a path like /cvmfs/mu2e.opensciencegrid.org/bin/muenv.sh

Hi Ray. 

- The reason `deactivate` gives permission denied is that the deactivate script is not set up with the execute permission, so you have to source it instead. 

- Got it, I've added the unset line. 

- I have added the full path in `setupmu2e-art.sh`. 

Thanks,
Sam

sam-grant commented:
I have made some changes in a recent commit, which also includes a name change to `pyenv` as suggested at the collaboration meeting.

rlcee commented:
did you want the function name to be mu2nev or pyenv?  It is still muenv in the setupmu2e-art

sam-grant commented:
Good catch. I changed the name.
